ocp_resources/pipeline.py

```diff
@@ -0,0 +1,55 @@
+# API reference: https://tekton.dev/docs/pipelines/pipelines/
+
+from ocp_resources.resource import NamespacedResource
+
+
+class Pipeline(NamespacedResource):
+    api_group = NamespacedResource.ApiGroup.TEKTON_DEV
+
+    def __init__(
+        self,
+        name=None,
+        namespace=None,
+        tasks=None,
+        params=None,
+        final_parallel_tasks=None,
+        client=None,
+        yaml_file=None,
+        **kwargs,
+    ):
+        """
+        Args:
+            name (str): Name of the pipeline
+            namespace (str): Namespace of the pipeline
+            tasks (str): Optional: actions to perform in pipeline
+            client: (DynamicClient): DynamicClient to use.
+            params (dict): Optional params to support pipelines.
+            params can be set/changed based on tasks.
+            Refer : https://tekton.dev/docs/pipelines/pipelines/#specifying-parameters
+            example: 'spec': {'params': [{'name': 'sourceTemplateName',
+            'type': 'string','default':'openshift'},
+            {'name': 'sourceTemplateNamespace', 'type':
+            'string', 'description': 'Namespace pf template'}]}
+            final_parallel_tasks: Optional: list of one or more final tasks that get executed
+            in parallel. This is added under finally in spec.
+        """
+        super().__init__(
+            name=name,
+            namespace=namespace,
+            client=client,
+            **kwargs,
+        )
+        self.tasks = tasks
```

Comment:
tasks is mandatory, please verify that the value is passed

Reply:
Yes document said it's mandatory but pipeline works without tasks as well. Raised a bug in tekton for it

Reply:
you are right.task should be important and if not present, we should see some sort of error. I will add a TODO

Reply:
tasks are optional based on current implementation.

Reply:
Is this TODO needed? you check if tasks passed in line 48

Reply:
we added TODO so that when bug is fixed we revisit the code and see what all changes we should made based on fix.
what do you think.